### PR TITLE
fix/user-name-fallbacks (PRO-667) -- Ready for Re-review and Merge

### PR DESCRIPTION
**Pull request by jonathanprozzi**
## Linear Ticket

- [PRO-667](https://linear.app/govrn/issue/PRO-667/display-nameuser-name-fallbacks)

## Description

- Adds truncated address fallbacks when a user name is null

## Screencaptures

Before:

attestation-no-name
table-no-name
home-shell-no-name
dashboard-no-name
govrn-attestation-dao-name

After:

table-name-fallback
attestation-name-fallback
dashboard-name-fallback
home-shell-name-fallback

### CODE REVIEW

File: apps/protocol-frontend/src/components/HomeShell.tsx [hunk 1]
```diff
@@ -86,7 +86,9 @@ const HomeShell = () => {
                   bgGradient="linear(to-l, #7928CA, #FF0080)"
                   bgClip="text"
                 >
-                  {userData?.name || userDataByAddress?.name}
+                  {userData?.name ||
+                    userDataByAddress?.name ||
+                    displayAddress(userData?.address)}
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/HomeShell.tsx:91]: I think we should abstract this or statement into a function. I can imagine us adding more logic such as ENS

Comment (jonathanprozzi) [reply apps/protocol-frontend/src/components/HomeShell.tsx:91]: Good call. Maybe we can do the fallbacks like this:
User name > ens > truncated address 

Will make a function we can reuse or adjust

**Comment by alexkeating** [comment]
Is this good or are we waiting for the ENS functionality?

**Comment by jonathanprozzi** [comment]
> Is this good or are we waiting for the ENS functionality?

We're waiting. Going to wrap it today

**Comment by jonathanprozzi** [comment]
Ah, wagmi's `useEnsName` hook doesn't support Gnosis out of the box. Will take a look a bit deeper here.

**Comment by alexkeating** [comment]
> Ah, wagmi's `useEnsName` hook doesn't support Gnosis out of the box. Will take a look a bit deeper here.

I think that makes sense, ENS isn't on Gnosis. Maybe, we need to create a mainnet provider to use that hook?

File: apps/protocol-frontend/src/components/ConnectWallet.tsx [hunk 2]
```diff
@@ -112,7 +117,7 @@ const ConnectWallet: React.FC<ConnectWalletProps> = ({ showNetwork }) => {
                       paddingRight={2}
                     >
                       <Text color="gray.800" fontSize="sm">
-                        {name || displayAddress(address || '')}
+                        {userEnsName ?? displayAddress(address)}
```
Comment (scottrepreneur) [inline apps/protocol-frontend/src/components/ConnectWallet.tsx:120]: can you use the hook here too?

**Review by scottrepreneur — APPROVED** [review]

**Comment by jonathanprozzi** [comment]
Made the following changes since review and approval:

- https://linear.app/govrn/issue/PRO-777/remove-mainnet-from-wagmi-for-now
- https://linear.app/govrn/issue/PRO-776/use-ens-fetch-outside-wagmi

- Adds in a `useEnsNameEthers` hook that is similar to how wagmi's `useEnsName` works under the hood (with react-query)
- This is utilized in our `useDisplayName` but also used separately in the `ConnectWallet` component since that only uses account or ENS, not the user's display name
- Having this separate allows for us to integrate it into other places like the table where we may want address or ENS only, and not the full display name fallbacks

File: apps/protocol-frontend/src/components/HomeShell.tsx [hunk 3]
```diff
@@ -80,7 +82,10 @@ const HomeShell = () => {
               <Text paddingBottom={8}>
                 Welcome back,{' '}
                 <Text as="span" fontWeight="bolder">
-                  {userData?.name || userDataByAddress?.name}
+                  {displayName}
+                  {/* {userData?.name ||
+                    userDataByAddress?.name ||
+                    displayAddress(userData?.address)} */}
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/HomeShell.tsx:88]: Left over comments

Comment (jonathanprozzi) [reply apps/protocol-frontend/src/components/HomeShell.tsx:88]: Ah! Thanks 🙏

File: apps/protocol-frontend/src/utils/web3.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@ const gnosisChain: Chain = {
 
 const dev = import.meta.env.MODE || false;
 const defaultChains = dev
-  ? [gnosisChain, chain.goerli, chain.rinkeby, chain.localhost]
+  ? [gnosisChain, chain.mainnet, chain.goerli, chain.rinkeby, chain.localhost]
```
Comment (alexkeating) [inline apps/protocol-frontend/src/utils/web3.ts:30]: Don't we need mainnet when dev is false?

Comment (jonathanprozzi) [reply apps/protocol-frontend/src/utils/web3.ts:30]: Should also remove it from dev if we aren't supporting

Comment (jonathanprozzi) [reply apps/protocol-frontend/src/utils/web3.ts:30]: I added in the `chain.mainnet` in the prior version of this PR when we used the wagmi hook for ENS. Removing this won't cause confusion in the interim and still lets use ENS

**Pull request merged**